guides/source/action_mailer_basics.md

```diff
@@ -468,181 +447,220 @@ class UserMailer < ApplicationMailer
-layout.
+```html+erb
+<%= link_to 'welcome', welcome_path %>
+```
 
-To use a different file, call [`layout`][] in your mailer:
+Use this:
 
-```ruby
-class UserMailer < ApplicationMailer
-  layout 'awesome' # use awesome.(html|text).erb as the layout
-end
+```html+erb
+<%= link_to 'welcome', welcome_url %>
 ```
 
-Just like with controller views, use `yield` to render the view inside the
-layout.
+By using the full URL, your links will work correctly in your emails.
 
-You can also pass in a `layout: 'layout_name'` option to the render call inside
-the format block to specify different layouts for different formats:
+#### Generating URLs with `url_for`
 
-```ruby
-class UserMailer < ApplicationMailer
-  def welcome_email
-    mail(to: params[:user].email) do |format|
-      format.html { render layout: 'my_layout' }
-      format.text
-    end
-  end
-end
+The [`url_for`][] helper generates a full URL by default in templates.
+
+If you haven't configured the `:host` option globally, you'll need to pass it to
+`url_for`.
+
+```erb
+<%= url_for(host: 'example.com',
+            controller: 'welcome',
+            action: 'greeting') %>
 ```
 
-Will render the HTML part using the `my_layout.html.erb` file and the text part
-with the usual `user_mailer.text.erb` file if it exists.
+[`url_for`]: https://api.rubyonrails.org/classes/ActionView/RoutingUrlFor.html#method-i-url_for
 
-[`layout`]: https://api.rubyonrails.org/classes/ActionView/Layouts/ClassMethods.html#method-i-layout
+#### Generating URLs with Named Routes
 
-### Previewing Emails
+Similar to other URLs, you need to use the `*_url` variant of named route
+helpers in emails as well.
 
-Action Mailer previews provide a way to see how emails look by visiting a
-special URL that renders them. In the above example, the preview class for
-`UserMailer` should be named `UserMailerPreview` and located in
-`test/mailers/previews/user_mailer_preview.rb`. To see the preview of
-`welcome_email`, implement a method that has the same name and call
-`UserMailer.welcome_email`:
+You either configure the `:host` option globally or make sure to pass it to the
+URL helper:
 
-```ruby
-class UserMailerPreview < ActionMailer::Preview
-  def welcome_email
-    UserMailer.with(user: User.first).welcome_email
-  end
-end
+```erb
+<%= user_url(@user, host: 'example.com') %>
 ```
 
-Then the preview will be available in <http://localhost:3000/rails/mailers/user_mailer/welcome_email>.
+### Adding Images in Action Mailer Views
 
-If you change something in `app/views/user_mailer/welcome_email.html.erb`
-or the mailer itself, it'll automatically reload and render it so you can
-visually see the new style instantly. A list of previews are also available
-in <http://localhost:3000/rails/mailers>.
+In order to use the `image_tag` helper in emails, you need to specify the
+`:asset_host` parameter. This is again because a mailer instance doesn't have
```

Comment:
the 'again because' might not make sense here if people are reading the sections in isolation. 

How about something like 'This is because a mailer instance doesn't have any context about the incoming request (as above).